worker/ecdsa/default_worker_prework.go

```
@@ -263,7 +264,21 @@ func (w *DefaultWorker) onPreExecutionResponse(tssMsg *commonTypes.TssMessage) e
 // We either start execution or finish this work.
 func (w *DefaultWorker) memberFinalized(msg *common.PreExecOutputMessage) {
 	if msg.Success {
-		// TODO: Check validity of the presign ids. Make sure it is never used.
+		if len(msg.PresignIds) > 0 {
+			statuses, err := w.db.LoadPresignStatus(msg.PresignIds)
+			if err != nil {
+				log.Error("error when loading presign data, presign ids =  ", msg.PresignIds)
+				return
+			}
+
+			for i, status := range statuses {
+				if status == db.PresignStatusUsed {
+					log.Warn("presign id was used. Id = ", msg.PresignIds[i])
+					return
```

Make failure callback instead of silently return here